tests/integration/__tests__/plugin-resolution.js

```diff
@@ -93,6 +93,55 @@ describe("automatically loads '@prettier/plugin-*' from --plugin-search-dir (dif
   });
 });
 
+describe("automatically loads '@<name>/prettier-plugin-*' from --plugin-search-dir (different to autoload dir)", () => {
+  runPrettier("plugins", [
+    "automatic/file.txt",
+    "--parser=foobar",
+    "--plugin-search-dir=automatic",
+  ]).test({
+    stdout: "foobar+contents" + EOL,
+    stderr: "",
+    status: 0,
+    write: [],
+  });
+});
+
+describe("automatically loads 'prettier-plugin-*' from pnpm virtual store", () => {
+  runPrettier("plugins/pnpmVirtualStore/prettier@0.0.0", [
+    "file.txt",
+    "--parser=foo",
+  ]).test({
+    stdout: "foo+contents" + EOL,
+    stderr: "",
+    status: 0,
+    write: [],
+  });
+});
+
+describe("automatically loads '@prettier/plugin-*' from pnpm virtual store", () => {
+  runPrettier("plugins/pnpmVirtualStore/prettier@0.0.0", [
+    "file.txt",
+    "--parser=bar",
+  ]).test({
+    stdout: "bar+contents" + EOL,
+    stderr: "",
+    status: 0,
+    write: [],
+  });
+});
+
+describe("automatically loads '@<name>/prettier-plugin-*' from pnpm virtual store", () => {
+  runPrettier("plugins/pnpmVirtualStore/prettier@0.0.0", [
+    "file.txt",
+    "--parser=foobar",
+  ]).test({
+    stdout: "foobar+contents" + EOL,
+    stderr: "",
+    status: 0,
+    write: [],
+  });
+});
```

Comment:
I don't think these tests are correctly testing the added part. The plugin search logic always starts from Prettier cli, if we want test, we'll have to put Prettier into those dirs.

Reply:
I could be misunderstanding the code, but I think in `run-prettier.js` the cwd gets set properly in `run()`. Either way, if I log `autoLoadDir`, it is set to the expected directory, so it does seem prettier is starting from the right directory.